lib/msal-common/src/authority/TrustedAuthority.ts

```diff
@@ -0,0 +1,93 @@
+import { Constants } from "../utils/Constants";
+import { INetworkModule } from "../network/INetworkModule";
+import { ICloudInstanceDiscoveryResponse } from "./ICloudInstanceDiscoveryResponse";
+import { CloudDiscoveryMetadataType } from "./CloudDiscoveryMetadataType";
+import { ICloudDiscoveryMetadata } from "./ICloudDiscoveryMetadata";
+import { ClientConfigurationError } from "../error/ClientConfigurationError";
+
+export class TrustedAuthority {
+    private static CloudDiscoveryMetadata: CloudDiscoveryMetadataType = {};
+
+    /**
+     * Set the CloudDiscoveryMetadata object from knownAuthorities or cloudDiscoveryMetadata passed into the app config
+     * @param knownAuthorities 
+     * @param cloudDiscoveryMetadata
+     */
+    public static setTrustedAuthoritiesFromConfig(knownAuthorities: Array<string>, cloudDiscoveryMetadata: string): void {
+        if (!this.getTrustedHostList().length){
+            if (knownAuthorities.length && cloudDiscoveryMetadata) {
```

Comment:
`if (knownAuthorities.length > 0 && !StringUtils.isEmpty(cloudDiscoveryMetadata))`

I know it is redundant but it helps with readability